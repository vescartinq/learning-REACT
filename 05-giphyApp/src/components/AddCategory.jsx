import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(''); // ''

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  // Funci√≥n que se ejecuta al clickar enter (<form onSubmit={handleSubmit}></form>)
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((newInput) => [inputValue, ...newInput]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
