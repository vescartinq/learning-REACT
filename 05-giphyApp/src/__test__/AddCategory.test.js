import React from 'react';
// import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { AddCategory } from './../components/AddCategory';

describe('AddCategory', () => {
  let wrapper;
  const setCategories = () => {};

  beforeEach(() => {
    //   Title y URL son required en PropTypes

    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should render AddCategory component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change input box', () => {
    const input = wrapper.find('input');
    const value = 'Messi';

    input.simulate('change', { target: { value } }); //en las pruebas no es necesario onChange, solo change
  }); // e.target.value='Messi'
});
