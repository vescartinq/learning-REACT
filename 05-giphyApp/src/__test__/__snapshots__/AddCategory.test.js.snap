// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should render AddCategory component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
